plugin/wp-cli-login-server.php

```diff
@@ -287,25 +302,29 @@ private function abort(Exception $e)
      */
     private function magicKey()
     {
-        return self::OPTION . '/' . $this->publicKey;
+        // We need to hash the salt to produce a key that won't exceed the maximum of 64 bytes.
+        $key = sodium_crypto_generichash(wp_salt('auth'));
+        $bin_hash = sodium_crypto_generichash($this->publicKey, $key);
+
+        return self::OPTION . '/' . sodium_bin2base64($bin_hash, SODIUM_BASE64_VARIANT_URLSAFE);
     }
 
     /**
      * Build the signature to check against the private key for this request.
      *
-     * @param WP_User $user
-     * @param string $redirect_url
+     * @param Magic Login data.
      *
      * @return string
      */
-    private function signature(WP_User $user, $redirect_url)
+    private function signature(Magic $magic)
```

Comment:
Just a refactoring suggestion, 

The logic to build a signature from inputs could be extracted.